docs: add es tooling blog post (#15)
e18e/e18e: docs/blog/es-tooling.md

---
title: ES Tooling - community lead e18e-focused tooling
author:

>   - name: The e18e Contributors

Reviewer: Perhaps it would be good to give credit to yourself 😄 I imagine we could have other guest writers in the future and it would be good to give the same credit too.

Reviewer 2: Ya 💯
Maybe this one could be signed as James and Passle if he is also going to do one of the sections?

sidebar: false
date: 2024-07-22
head:
  - - meta
    - property: og:type
      content: website
  - - meta
    - property: og:title
      content: es tooling - community lead e18e-focused tooling
  - - meta
    - property: og:url
      content: https://e18e.dev/blog/es-tooling
  - - meta
    - property: og:description
      content: ES Tooling - a community lead, e18e-focused collection of tools for the ecosystem
---

# ES Tooling

_July 22, 2024_

> The e18e initiative has seen huge growth since launch. We've seen hundreds of performance contributions upstream to popular projects, and many community lead tools.

Reviewer 2: ```suggestion
The e18e initiative has seen huge growth since launch. We've seen hundreds of performance contributions upstream to popular projects, and the creation of many community lead tools.
```

> The e18e initiative has seen huge growth since launch. We've seen hundreds of performance contributions upstream to popular projects, and the creation of many community lead tools.

Reviewer 3: Should we add a short recap/summary of what exactly e18e is, and a link to the https://e18e.dev/guide/#why-e18e page? Maybe a lot of people will see this post not knowing what e18e is

A notable example of this is the [ES Tooling](https://github.com/orgs/es-tooling/repositories) community GitHub organisation.

The ES Tooling organisation has been created to hold various e18e-focused tools and forks maintained by the community.

## A home for e18e-focused tools

Over the last few months, many of us have collaborated and connected through the e18e community. The ES tooling organisation is a result of that - a place for us to store and own forks, tools, and other code bases we're working on together.

A few popular projects that were running solo until now, have come together to create this:

> - [ecosystem cleanup](https://github.com/43081j/ecosystem-cleanup).

Reviewer 3: Maybe a short summary of what these tools/are do, in 1 sentence?

Reviewer 3: nvm, I see we go deeper into them down below

- [module replacements](https://github.com/es-tooling/module-replacements).

> - [module replacements code mods](https://github.com/thepassle/module-replacements-codemods)

Reviewer 3: ```suggestion
- [module replacements codemods](https://github.com/thepassle/module-replacements-codemods)
```

- [fd-package-json](https://github.com/es-tooling/fd-package-json)

> 

Reviewer 2: Maybe we could add a paragraph about why these repos were not move to the e18e org?

```suggestion

The ES Tooling org predates the creation of the e18e initiative. In case you're wondering about why these efforts won't live under the e18e org, we think that it is better to keep this organization focused on [fostering the space to connect us](https://chat.e18e.dev/), and maintaining a curation of ecosystem performance projects.

```

## Ecosystem cleanup

One of the projects which kicked much of this initiative off - the ecosystem cleanup project has now moved into the ES Tooling organisation to allow better visibility and maintenance.

The cleanup project exists primarily as an issue tracker, tracking:

- Footprint reduction in popular packages
- Missing forks or alternatives that could be worked on
- Performance improvements in popular packages

If you want to help out, this project is a good starting point to find some open issues.

> Read more at the [ecosystem cleanup project](https://github.com/43081j/ecosystem-cleanup).

Reviewer 2: ```suggestion
Read more at the [ecosystem cleanup project](https://github.com/es-tooling/ecosystem-cleanup).
```

## Module replacements

> The module replacements project is a community lead list of popular NPM packages and their possible replacements.

Reviewer 4: ```suggestion
The module replacements project is a community lead list of popular npm packages and their possible replacements.
```

Often, these replacements are native, or much faster and more modern packages. If you know of a good alternative to something, get involved and add to the list!

Read more at the [module replacements project](https://github.com/es-tooling/module-replacements).

> ## Module replacements code mods

Reviewer 3: ```suggestion
## Module replacements codemods
```

---
title: ES Tooling - community lead e18e-focused tooling
author:

>   - name: James Garbutt and Pascal Schilp

Reviewer: Note: We should probably figure out how to support multiple authors in the future.

sidebar: false
date: 2024-07-22
head:
  - - meta
    - property: og:type
      content: website
  - - meta
    - property: og:title

>       content: es tooling - community lead e18e-focused tooling

Reviewer: ```suggestion
      content: ES tooling - community lead e18e-focused tooling
```

The casing should probably match the title.

  - - meta
    - property: og:url
      content: https://e18e.dev/blog/es-tooling
  - - meta
    - property: og:description
      content: ES Tooling - a community lead, e18e-focused collection of tools for the ecosystem
---

# ES Tooling

_July 22, 2024_

The e18e initiative has seen huge growth since launch. We've seen hundreds of performance contributions upstream to popular projects, and the creation of many community lead tools.

> A notable example of this is the [ES Tooling](https://github.com/orgs/es-tooling/repositories) community GitHub organisation.
> 
> The ES Tooling organisation has been created to hold various e18e-focused tools and forks maintained by the community.

Reviewer: ```suggestion
A notable example of this is the [ES Tooling](https://github.com/orgs/es-tooling/repositories) community GitHub organisation, created to hold various e18e-focused tools.
```

Maybe we can simply this part so the next section further clarifies what the es tooling org is about.

If you haven't yet encountered e18e, read more in the [guide](https://e18e.dev/guide/#why-e18e) and [join the community](https://chat.e18e.dev/).

## A home for e18e-focused tools

Over the last few months, many of us have collaborated and connected through the e18e community. The ES tooling organisation is a result of that - a place for us to store and own forks, tools, and other code bases we're working on together.

A few popular projects that were running solo until now, have come together to create this:

> - [ecosystem cleanup](https://github.com/43081j/ecosystem-cleanup).
> - [module replacements](https://github.com/es-tooling/module-replacements).
> - [module replacements codemods](https://github.com/thepassle/module-replacements-codemods)
> - [fd-package-json](https://github.com/es-tooling/fd-package-json)

Reviewer: ```suggestion
- [ecosystem-cleanup](https://github.com/43081j/ecosystem-cleanup).
- [module-replacements](https://github.com/es-tooling/module-replacements).
- [module-replacements-codemods](https://github.com/thepassle/module-replacements-codemods)
- [fd-package-json](https://github.com/es-tooling/fd-package-json)
```

Maybe we should hyphenate so it's consistent.

Author: these were meant to be the readable names rather than the url slugs

i suppose since the npm packages have hyphens, we could though

> The ES Tooling org predates the creation of the e18e initiative. In case you're wondering about why these efforts won't live under the e18e org, we think that it is better to keep this organization focused on [fostering the space to connect us](https://chat.e18e.dev/), and maintaining a curation of ecosystem performance projects.

Reviewer: I'm a bit confused of the `fostering the space to connect us` part. Is this referring to e18e or es tooling? I assume the latter as before it mentions "this organization focused on" which I assume refers to es tooling.

If so, I thought e18e is the place for `fostering the space to connect us` and not directly related to "this organization"?

Author: this was a previous PR comment i just committed. better to ask @patak-dev 

he was trying to point out why this stuff doesn't live in e18e's github org

Reviewer 2: Is this more clear?
```suggestion
The ES Tooling org predates the creation of the e18e initiative. In case you're wondering about why these efforts won't live under the e18e org, we think that it is better to keep the e18e organization focused on [fostering a space to connect us](https://chat.e18e.dev/), and as a way to curate ecosystem performance projects through these docs, blog, and social accounts.
```

## Module replacements

The module replacements project is a community lead list of popular npm packages and their possible replacements.

Often, these replacements are native, or much faster and more modern packages. If you know of a good alternative to something, get involved and add to the list!

Read more at the [module replacements project](https://github.com/es-tooling/module-replacements).

## Module replacements codemods

The module replacements codemods project is an effort to automate the ecosystem cleanup. Taking the module replacements as input, we aim to have automated codemods for all packages listed. This way other tooling can use these codemods to perform transformations on projects automatically. For example, a codemod for the `is-even` package would result in:

Before:
```js
const isEven = require('is-even')
console.log(isEven(2))
```

After:
```js
console.log(2 % 2 === 0)
```

We'll also be looking into implementing a CLI around these codemods so people can run them on their own projects, or so they can run them on other projects and create pull requests to them, hopefully simplifying the amount of effort it takes to replace dependencies.

> You can find more information about the [module replacements codemods](https://github.com/es-tooling/module-replacements-codemods) repository. If you're interested in seeing the before/after states of the codemods, take a look at the [test fixtures](https://github.com/es-tooling/module-replacements-codemods/tree/main/test/fixtures), every codemod has one or more before/after example available.

Reviewer: ```suggestion
You can find more information about the [module replacements codemods](https://github.com/es-tooling/module-replacements-codemods) repository. If you're interested in seeing the before/after states of the codemods, take a look at the [test fixtures](https://github.com/es-tooling/module-replacements-codemods/tree/main/test/fixtures). Every codemod has one or more before/after examples available.
```
